main.tf

```diff
@@ -1,13 +1,89 @@
-resource "random_integer" "example" {
-  count = module.this.enabled ? 1 : 0
+#-----------------------------------------------------------------------------------------------------------------------
+# Subscribe the Acccount to GuardDuty
+#-----------------------------------------------------------------------------------------------------------------------
+resource "aws_guardduty_detector" "guardduty" {
+  enable                       = module.this.enabled
+  finding_publishing_frequency = var.finding_publishing_frequency
+}
+
+#-----------------------------------------------------------------------------------------------------------------------
+# Optionally configure Event Bridge Rules and SNS subscriptions 
+# https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-cwe-integration-types.html
+# https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/resource-based-policies-cwe.html#sns-permissions
+#-----------------------------------------------------------------------------------------------------------------------
+module "sns_topic" {
+
+  source  = "cloudposse/sns-topic/aws"
+  version = "0.9.0"
+  count   = local.create_sns_topic ? 1 : 0
+
+  attributes      = ["guardduty"]
+  subscribers     = var.subscribers
+  sqs_dlq_enabled = false
+
+  context = module.this.context
+}
+
+module "findings_label" {
+  source  = "cloudposse/label/null"
+  version = "0.21.0"
+
+  attributes = ["guardduty-findings"]
+  context    = module.this.context
+}
+
+resource "aws_sns_topic_policy" "sns_topic_publish_policy" {
+  count  = module.this.enabled && local.create_sns_topic ? 1 : 0
+  arn    = local.findings_notification_arn
+  policy = data.aws_iam_policy_document.sns_topic_policy[0].json
+}
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
+data "aws_iam_policy_document" "sns_topic_policy" {
+  count     = module.this.enabled && local.create_sns_topic ? 1 : 0
+  policy_id = "GuardDutyPublishToSNS"
+  statement {
+    sid = ""
+    actions = [
+      "sns:Publish"
+    ]
+    principals {
+      type        = "Service"
+      identifiers = ["cloudwatch.amazonaws.com"]
+    }
+    resources = [module.sns_topic[0].sns_topic.arn]
+    effect    = "Allow"
   }
 }
 
+resource "aws_cloudwatch_event_rule" "findings" {
```

Comment:
@aknysh @mcalhoun does it make sense to have a `catalog/` of event rules similar to we do for SCP/datadog/etc?

Reply:
if those rules can be configured in YAML, then yes, `catalog` would be nice and would follow the same pattern we use in:

https://github.com/cloudposse/terraform-aws-service-control-policies/tree/master/catalog

https://github.com/cloudposse/terraform-datadog-monitor/tree/master/catalog

We should use `terraform-yaml-config` to read the from YAML catalog:
https://github.com/cloudposse/terraform-aws-service-control-policies/blob/master/examples/complete/main.tf#L2

Reply:
There aren't really any configurable event rules for Guard Duty, the only config that would be applicable to this comment would be a catalog of SNS subscriptions, but probably doesn't make sense as the variability in params would make it difficult.